pygmt/figure.py

```diff
@@ -328,11 +339,14 @@ def savefig(
         if anti_alias:
             kwargs["Qt"] = 2
             kwargs["Qg"] = 2
-        if ext == "kml":
-            kwargs["W"] = "+k"
 
         self.psconvert(prefix=prefix, fmt=fmt, crop=crop, **kwargs)
 
+        # Remove the .pgw world file if exists
+        # See upstream fix https://github.com/GenericMappingTools/gmt/pull/7865
+        if ext == "tiff" and fname.with_suffix(".pgw").exists():
+            fname.with_suffix(".pgw").unlink()
```

Comment:
Not sure how many years into the future we'll set GMT 6.5.0 as the minimum required version, but let's add a note as a reminder. Feel free to reword before commiting.

```suggestion
        # Remove the .pgw world file if exists (remove when PyGMT sets minimum
        # required GMT version to 6.5.0).
        # See upstream fix https://github.com/GenericMappingTools/gmt/pull/7865
        if ext == "tiff" and fname.with_suffix(".pgw").exists():
            fname.with_suffix(".pgw").unlink()
```